src/main/clojure/clara/rules/compiler.clj

```diff
@@ -1721,6 +1732,7 @@
         ;;
         ;; Note that this ordering is not for correctness; we are just trying to increase consistency of rulebase compilation,
         ;; and hopefully thereby execution times, from run to run.
+        productions (names-unique productions)
```

Comment:
The output of names-unique isn't really useful since it just returns its input or throws an exception, but someone reading through this code would need to look at the implementation of names-unique to know that.  Perhaps just 

````
_ (names-unique productions)
````
This currently doesn't actually change the value of the "productions" binding later but it looks like it could, which is unnecessary obfuscation IMO.

Reply:
I think it is fairly idiomatic to have functions named like `validate-x` that just returned `x` unchanged or threw an exception if invalid.

It may still make sense to bind to underscore (ie ignored) if that helps readability though.

Reply:
Done.